Sources/AutobahnDescription/AutobahnDescription.swift

```diff
@@ -60,37 +62,41 @@ public final class Autobahn<H: Highway> {
     
     // MARK: - Public Functions
     
-    public init(_ highwayType: H.Type) {}
+    public init(_ highwayType: H.Type) {
+        self.highwayStart = Date()
+    }
     
     public func beforeAll(handler: @escaping (H) throws -> Void) -> Autobahn<H> {
-        guard self.beforeAllHandler == nil else {
-            fatalError("beforeAll declared more than once")
-        }
-        self.beforeAllHandler = handler
+        precondition(beforeAllHandler == nil, "beforeAll declared more than once")
+        beforeAllHandler = handler
         return self
     }
     
-    public func highway(_ highway: H, dependsOn: [H]? = nil, taskHandler: @escaping () throws -> Void) -> Autobahn<H> {
-        self.highways[highway] = taskHandler
-        if let d = dependsOn {
-            self.dependings[highway] = d
-        }
+    public func highway(_ highway: H, handler: @escaping () throws -> Void) -> Autobahn<H> {
+        self.highways[highway] = handler
+        return self
+    }
+
+    public func highway(_ highway: H, dependsOn highways: [H]) -> Autobahn<H> {
+        dependings[highway] = highways
+        return self
+    }
+
+    public func highway(_ highway: H, dependsOn highways: [H], handler: @escaping () throws -> Void) -> Autobahn<H> {
+        self.highways[highway] = handler
+        dependings[highway] = highways
         return self
     }
     
     public func afterAll(handler: @escaping (H) throws -> Void) -> Autobahn<H> {
-        guard self.afterAllHandler == nil else {
-            fatalError("afterAll declared more than once")
-        }
-        self.afterAllHandler = handler
+        precondition(afterAllHandler == nil, "afterAll declared more than once")
```

Comment:
I also thought about not prohibiting the definition of multiple afterAll handlers. They could also be stored in an array and be called in order they are defined. What do you think?

Reply:
What would be the use case for wanting to define multiple `afterAll` closures?  One may be to split logic up into sections which can also be done by defining your own methods in the `Autobahn.swift` and calling them in the the `afterAll`

Reply:
I just think we should not prohibit it if it makes no technical difference. Currently it's a function wrapped in an Optional, as an array it would be wrapped in an Array. In both cases we need to unwrap, using `if let` or `for in`. So it really doesn't matter but restricts the API with a precondition crash if used 'wrong'.

Reply:
I would prefer to wait until we have some requests to add this ability rather than add this feature now and feel like it should be removed at a later point which would be a breaking change